common/task/parallelTaskProcessor.go

```diff
@@ -152,6 +155,13 @@ func (p *parallelTaskProcessorImpl) executeTask(task Task, shutdownCh chan struc
 	sw := p.metricsScope.StartTimer(metrics.ParallelTaskTaskProcessingLatency)
 	defer sw.Stop()
 
+	defer func() {
+		if r := recover(); r != nil {
+			p.logger.Error("recovered panic in task execution", tag.Dynamic("recovered-panic", r))
+			task.HandleErr(fmt.Errorf("recovered panic: %v", r))
```

Comment:
Might also need to nack it?  I assume `HandleErr` doesn't, since it's part of the retried `op` and ack/nack is done externally after retries.

Reply:
I thought about that and honestly wasn't too sure,  it's not clear to me that all panics are non-retriable (they could be some transient error causing data to be mishandled) and I was going to err on the more cautious just-log-the-thing initially

Reply:
you should either nack or ack it, but doing nothing is not correct

Reply:
ok, I'll try nack